.github/workflows/main.yaml

```diff
@@ -0,0 +1,41 @@
+name: Main
+
+on:
+  push:
+    branches: [master, staging, dev]
+  pull_request:
+    types: [opened, edited, reopened, synchronize, ready_for_review, unlocked, auto_merge_enabled]
+    branches: [master, staging, dev]
+
+jobs:
+  main:
+    name: Quality validation
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Node.js (v16)
+        uses: actions/setup-node@v2
+        with:
+          node-version: "16"
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn config get cacheFolder)"
```

Comment:
What do these `::` do?
Does just `echo`-ing something does anything?

Reply:
This echo is used in the next step. But after couple of hours of testing this cache strategy from guides I saw no results, so will remove these two steps. Another option is to cache directly node_modules, but think this can be done later if necessary, as not recommended.

Reply:
If about :: - https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions